src/Tasks/Microsoft.NET.Build.Tasks/PreprocessPackageDependenciesDesignTime.cs

```diff
@@ -182,6 +191,82 @@ private void PopulateAssemblies()
             }
         }
 
+        /// <summary>
+        /// Update FileDefinitions and FileDependencies to pretend that certain Reference items
+        /// explicitly added by a .targets file in the NETStandard.Library are actually normal
+        /// package items. This allows them to show up properly under the "SDK" node in Solution
+        /// Explorer, rather than just under the "Assemblies" node.
+        ///
+        /// This is not meant to be a general mechanism for injecting files that can't be handled
+        /// through the normal NuGet package mechanisms. For that, package authors should be
+        /// creating FileDefinition and FileDependency items instead of Reference items.
+        /// </summary>
+        private void PopulatePreResolvedNetStandardAssemblies()
+        {
+            var preResolvedAssemblyDependencies = new List<ITaskItem>();
+            foreach (var reference in References)
+            {
+                var packageName = reference.GetMetadata("NuGetPackageId");
+                var packageVersion = reference.GetMetadata("NuGetPackageVersion");
+
+                // This is not a "pre-resolved" assembly; skip it.
+                if (packageName == null || packageVersion == null)
+                {
+                    continue;
+                }
+
+                // This is meant only as a workaround for an issue in NETStandard.Library 2.0.*.
+                // Other packages should use the more general mechanism of creating
+                // FileDefinition and FileDependency items rather than Reference items.
+                if (packageName != "NETStandard.Library")
+                {
+                    continue;
+                }
```

Comment:
Let me expand a bit on that second issue.

Say you are a package author and can't add references from the assets file for whatever reason so you need to use this mechanism. Say you also need to include different files for netstandard2.0 and net46. You'll need to do something like this:

``` xml
<Reference Include="net46support\Alpha.dll" Condition="'$(TargetFramework)' == 'net46'">
  <NuGetPackageId>MyPackage</NuGetPackageId>
  <NuGetPackageVersion>1.0.0</NuGetPackageVersion>
  <NuGetTargetFramework>.Net Framework,Version=4.6</NuGetTargetFramework>
</Reference>
<Reference Include="netstandard20support\Beta.dll" Condition="'$(TargetFramework)' == 'netstandard2.0'">
  <NuGetPackageId>MyPackage</NuGetPackageId>
  <NuGetPackageVersion>1.0.0</NuGetPackageVersion>
  <NuGetTargetFramework>.Net Standard,Version=2.0</NuGetTargetFramework>
</Reference>
```

Is the project system even going to see both references, or only one at a time depending on the "active" target framework? The end result being that in multi-targeting scenarios you might only see one or the other assembly in Solution Explorer, not both.

We don't have this problem with `FileDefinition` and `FileDependency` items.